cmd/notification.go

```diff
@@ -452,6 +470,7 @@ func NewNotificationSys(config *serverConfig, endpoints EndpointList) *Notificat
 		bucketRulesMap:             make(map[string]event.RulesMap),
 		bucketRemoteTargetRulesMap: make(map[string]map[event.TargetID]event.RulesMap),
 		peerRPCClientMap:           peerRPCClientMap,
+		mode:                       serverMode,
```

Comment:
I think user spilling (gateway or server) is not a good idea.  This can be avoided by

1. Supply `targetList` as arg than `config` to this function.
2. In `NotificationSys.Init()`, you could pass additional argument as `notifyAll bool = true` for gateways.
3. In `NotificationSys.Send()` have `targetIDs` as per `notifyAll` flag i.e. `MINIO_GATEWAY_NOTIFY` value in this case.